securedrop/journalist_app/sessions.py

```diff
@@ -0,0 +1,232 @@
+import typing
+from datetime import datetime, timedelta, timezone
+from secrets import token_urlsafe
+from flask import Flask, Request, Response
+from flask import current_app as app
+from typing import Optional, Any
+from flask.sessions import SessionInterface as FlaskSessionInterface
+from flask.sessions import SessionMixin, session_json_serializer
+from json.decoder import JSONDecodeError
+from redis import Redis
+from werkzeug.datastructures import CallbackDict
+from itsdangerous import URLSafeTimedSerializer, BadSignature
+
+from models import Journalist
+
+
+class ServerSideSession(CallbackDict, SessionMixin):
+    """Baseclass for server-side based sessions."""
+
+    def __init__(self, sid: str, token: str, lifetime: int = 0, initial: Any = None) -> None:
+        def on_update(self: ServerSideSession) -> None:
+            self.modified = True
+        if initial and 'uid' in initial:
+            self.set_uid(initial['uid'])
+            self.set_user()
+        CallbackDict.__init__(self, initial, on_update)
+        self.sid = sid
+        self.token: str = token
+        self.lifetime = lifetime
+        self.is_api = False
+        self.to_destroy = False
+        self.to_regenerate = False
+        self.modified = False
+
+    def get_token(self) -> Optional[str]:
+        return self.token
+
+    def get_lifetime(self) -> datetime:
+        return datetime.now(timezone.utc) + timedelta(seconds=self.lifetime)
+
+    def set_user(self) -> None:
+        if hasattr(self, 'uid') and self.uid is not None:
+            self.user = Journalist.query.get(self.uid)
+
+    def get_user(self) -> Optional[Journalist]:
+        if hasattr(self, 'user'):
```

Comment:
In general I don't like dynamic properties like this, where they might not exist, it makes static typing harder. Why not have the variable exist from the beginning, but set to `None`, so it's just typed as `Optional[Journalist]` and can be returned directly? 

Same applies to `self.uid` too.